### PR TITLE
change: valuename from fields to netflow_fields

### PR DESCRIPTION
**Pull request by t-umeno**
reason:
unexpected error error="field 'fields' is a reserved name in BinData::Array" #18
https://github.com/repeatedly/fluent-plugin-netflow/issues/18

Signed-off-by: Takashi Umeno <umeno.takashi@gmail.com>

### CODE REVIEW

File: lib/fluent/plugin/netflow_records.rb [hunk 1]
```diff
@@ -112,7 +112,7 @@ class TemplateFlowset < BinData::Record
         array  :templates, read_until: lambda { array.num_bytes == flowset_length - 4 } do
           uint16 :template_id
           uint16 :field_count
-          array  :fields, initial_length: :field_count do
+          array  :netflow_fields, initial_length: :field_count do
```
Comment (codeout) [inline lib/fluent/plugin/netflow_records.rb]: How about ```:template_fields```?

```template.netflow_fields``` sounds something misleading... ```netflow_fields``` reminds me of data flowset, not template flowset.

https://github.com/t-umeno/fluent-plugin-netflow/blob/2e90a42cc18c996c734f43b192ee20768f97607c/lib/fluent/plugin/parser_netflow.rb#L205

```
            template.netflow_fields.each do |field|
```

File: lib/fluent/plugin/parser_netflow.rb [hunk 2]
```diff
@@ -31,7 +31,7 @@ def configure(conf)
         filename = File.expand_path('../netflow_fields.yaml', __FILE__)
 
         begin
-          @fields = YAML.load_file(filename)
+          @netflow_fields = YAML.load_file(filename)
```
Comment (codeout) [inline lib/fluent/plugin/parser_netflow.rb]: I have no idea, but should we modify local / instance variable names according to BinData change?

File: lib/fluent/plugin/netflow_records.rb (hunk 1 above)
Comment (t-umeno) [inline lib/fluent/plugin/netflow_records.rb]: I think :template_fields is more better.

File: lib/fluent/plugin/parser_netflow.rb (hunk 2 above)
Comment (t-umeno) [inline lib/fluent/plugin/parser_netflow.rb]: This line change is not needed. Please use @fields.

**Comment by repeatedly** [comment]
I updated .travis.yml. Could you rebase your commits?

**Comment by repeatedly** [comment]
After applied codeout's review, I will merge this patch.

**Comment by repeatedly** [comment]
Thanks!

**Comment by t-umeno** [comment]
Thanks!

2016-07-28 10:00 GMT+09:00 Masahiro Nakagawa <notifications@github.com>:

**Pull request merged**